src/features/governance/updateProposalsInDB.ts

```diff
@@ -145,11 +145,25 @@ async function mergeProposalDataIntoPGRow({
   );
   let mostRecentProposalState = await getProposalOnChain(client, proposalId);
   if (BigInt(mostRecentProposalState[0]) === 0n) {
+    const lastVoteEvent = (
+      await database
+        .select()
+        .from(eventsTable)
+        .where(
+          and(
+            eq(eventsTable.eventName, 'ProposalVotedV2'),
+            eq(sql`(${eventsTable.args}->>'proposalId')::bigint`, proposalId),
+          ),
+        )
+        .orderBy(desc(eventsTable.blockNumber))
+        .limit(1)
+    )[0];
+
     // we can't rely on events as they don't all contain timestamps
     mostRecentProposalState = await getProposalOnChain(
       client,
       proposalId,
-      BigInt(lastProposalEvent.blockNumber) - 1n,
+      lastVoteEvent?.blockNumber || BigInt(lastProposalEvent.blockNumber),
```

Comment:
I think what we actually want here is the higher blocknumber.

some cases

1. a proposal is dequeued and voted on (use vote event aka high block number)
2. a proposal is dequeued and voted on and approved (use higher block number) 
3. a proposal s dequeued and voted on and approved and executed (use exectuted event -- high block number)


@nicolasbrugneaux